api/app/controllers/spree/api/v2/storefront/stores_controller.rb

```diff
@@ -7,8 +7,27 @@ def current
             render_serialized_payload { serialize_resource(current_store) }
           end
 
+          def export_google_rss_feed
+            settings
+            if @settings.nil?
+              raise ActionController::RoutingError, 'Not Found'
+            elsif @settings.enabled
+              send_data export_google_rss_service.value[:file], filename: 'products.rss', type: 'text/xml'
+            else
+              send_data 'Export not enabled for those settings.'
+            end
+          end
+
           private
 
+          def settings
+            @settings = Spree::DataFeedSetting.find_by(spree_store_id: current_store, uuid: params[:unique_url])
```

Comment:
```suggestion
            @settings ||= Spree::DataFeedSetting.find_by!(spree_store_id: current_store, uuid: params[:unique_url], enabled: true)
```

You can then skip the additional verification of whether a feed is enabled, that you did in the `show` method.

Reply:
It also runs the db query only the first time someone accesses it, so you can then call `settings` in any place you need them, like an `attr_accessor`.